spring-rabbit-junit/src/test/java/org/springframework/amqp/rabbit/junit/BrokerRunningTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.amqp.rabbit.junit;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
+import org.springframework.beans.DirectFieldAccessor;
+
+import com.rabbitmq.client.ConnectionFactory;
+
+/**
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class BrokerRunningTests {
+
+	@Test
+	public void testVars() {
+		BrokerRunning brokerRunning = BrokerRunning.isBrokerAndManagementRunning();
+		brokerRunning.setAdminPassword("foo");
+		brokerRunning.setAdminUser("bar");
+		brokerRunning.setHostName("baz");
+		brokerRunning.setPassword("qux");
+		brokerRunning.setPort(1234);
+		brokerRunning.setUser("fiz");
+
+		assertEquals("http://baz:15672/api/", brokerRunning.getAdminUri());
+		ConnectionFactory connectionFactory = brokerRunning.getConnectionFactory();
+		assertEquals("baz", connectionFactory.getHost());
+		assertEquals(1234, connectionFactory.getPort());
+		assertEquals("fiz", connectionFactory.getUsername());
+		assertEquals("qux", connectionFactory.getPassword());
+	}
+
+	@Test
+	public void testEnvironmentVars() {
+		Map<String, String> vars = new HashMap<>();
+		vars.put("RABBITMQ_TEST_ADMIN_PASSWORD", "FOO");
+		vars.put("RABBITMQ_TEST_ADMIN_URI", "http://foo/bar");
+		vars.put("RABBITMQ_TEST_ADMIN_USER", "BAR");
+		vars.put("RABBITMQ_TEST_HOSTNAME", "BAZ");
+		vars.put("RABBITMQ_TEST_PASSWORD", "QUX");
+		vars.put("RABBITMQ_TEST_PORT", "2345");
+		vars.put("RABBITMQ_TEST_USER", "FIZ");
+		BrokerRunning.setEnvironmentVariables(vars);
+		BrokerRunning brokerRunning = BrokerRunning.isBrokerAndManagementRunning();
```

Comment:
I can agree about the point to set once, but how about to let to reset any way?

But at the same time I can't agree for your first comment.
Just swap these two lines and see how test works!

Looks what we have there:
```
public static BrokerRunning isRunning() {
	return new BrokerRunning(true);
}
```
So, we create a new instance and initialize its properties from the ENV, not `environmentOverrides`, because we don't have them yet.